### PR TITLE
chore(node): Optimize transport for memory and bandwitdh

### PR DESCRIPTION
**Pull request by oblique**

### CODE REVIEW

File: node/src/transport.rs [hunk 1]
```diff
@@ -21,20 +21,25 @@ mod imp {
...
-            yamux_config.set_max_buffer_size(3 * 1024 * 1024);
+            // This increases bandwidth utilitation. With 1MB of receive
+            // window, we can utilize 81.92mbps (10mb/s) per stream when
+            // latency is 100ms: 1mb / 100ms * 8bits = 81.92mbps
+            //
+            // This means that machine needs 8gb of ram to handle 8192
+            // streams (the default). For this reason we lower max streams
+            // to 2048.
+            //
+            // More info: https://github.com/libp2p/rust-yamux/issues/162
```
Comment (zvolin) [inline node/src/transport.rs:39]: wondering about also an info that Go impl has the max window as 16MB and dropping a link to
https://github.com/libp2p/go-libp2p/blob/master/p2p/muxer/yamux/transport.go#L24
leaving this up to you

**Review by zvolin — APPROVED** [review]
looks good

**Review by fl0rek — APPROVED** [review]

**Review by fl0rek — APPROVED** [review]
lgtm!

File: node/src/transport.rs (hunk 1 above)
Comment (oblique) [reply node/src/transport.rs:39]: done

**Review by zvolin — APPROVED** [review]

**Pull request merged**